### PR TITLE
Forbid usage of Literal[None] annotation.

### LINKED ISSUE #829: Forbid to use `Literal[None]`
# Rule request

## Thesis

`Literal[None]` is just the same as `None`. There's no need to use the first version.
It is not type related, it is a consistency rule.

### PR DESCRIPTION
**Pull request by lensvol**
Closes #829 

I've decided to fold the check into an existing one for the wrong annotations, although
it took some doing to appease `mypy`.

## Checklist

- [X] I have double checked that there are no unrelated changes in this pull request (old patches, accidental config files, etc)
- [X] I have created at least one test case for the changes I have made
- [X] I have updated the documentation for the changes I have made
- [X] I have added my changes to the `CHANGELOG.md`

## Related issues

On the side note, you may need to extend your contributor's guidelines to include recommendations on how to properly write integration tests in your project. Debugging those cryptic failures was somewhat fun, but rather non-trivial.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,13 @@ We follow Semantic Versions since the `0.1.0` release.
 We used to have incremental versioning before `0.1.0`.
 
 
+## 0.13.1
```
Comment (sobolevn) [inline CHANGELOG.md]: Oh, my bad. It should be in `## 0.13.0 WIP`. It is not yet released

File: wemake_python_styleguide/violations/consistency.py [hunk 2]
```diff
@@ -1748,3 +1750,34 @@ class OperationSignNegationViolation(ASTViolation):
 
     error_template = 'Found wrong operation sign'
     code = 346
+
+
+@final
+class LiteralNoneViolation(ASTViolation):
```
Comment (sobolevn) [inline wemake_python_styleguide/violations/consistency.py]: After creating so many violations about types - I have decided to make a special violation group and a visitor for them. Let's move this one into `violations/annotations.py` and use a new code (`WPS7xx`, I guess)

File: wemake_python_styleguide/visitors/ast/annotations.py [hunk 3]
```diff
@@ -42,13 +44,29 @@ def visit_arg(self, node: ast.arg) -> None:
         self._check_arg_annotation(node)
         self.generic_visit(node)
 
+    def _check_for_literal_none(self, node: ast.Subscript) -> None:
```
Comment (sobolevn) [inline wemake_python_styleguide/visitors/ast/annotations.py]: As I already said: let's move this logic into `visitors/ast/annotations.py`

Comment (lensvol) [reply wemake_python_styleguide/visitors/ast/annotations.py]: Done, please check the freshest commit. :)

**Comment by lensvol** [comment]
Implemented requested changes.

**Comment by sobolevn** [comment]
Thanks a lot, @lensvol!

**Pull request merged**